### PR TITLE
Update tests so that they don't rely on specific credentials

### PR DESCRIPTION
**Pull request by espoem**
Running tests locally required specific credentials for `fpl.login()` method. Mocking was introduced to abstract the logic of fetching data and a few methods' logic was fixed according to the docs.

### CODE REVIEW

File: tests/test_fixture.py [hunk 1]
```diff
@@ -1,40 +1,185 @@
...
+    "minutes": 90,
+    "provisional_start_time": False,
+    "finished_provisional": True,
+    "event": 1,
+    "team_a": 11,
+    "team_h": 14,
+}
+
+
+@pytest.fixture()
```
Comment (amosbastian) [inline tests/test_fixture.py]: Can't you use the fixture in `conftest.py`?

File: tests/test_h2h_league.py [hunk 2]
```diff
@@ -1,8 +1,91 @@
...
+        "admin_entry": None,
+        "start_event": 1,
+    },
+    "standings": {"has_next": False, "number": 1, "results": []},
+    "matches_next": {},
+    "matches_this": {},
+}
+
+
+@pytest.fixture()
```
Comment (amosbastian) [inline tests/test_h2h_league.py]: Same here, there is already a fixture for `h2h_league` in `conftest.py`.

File: tests/test_h2h_league.py [hunk 3]
```diff
@@ -1,8 +1,91 @@
...
+        for k, v in h2h_league_data.items():
+            assert getattr(league, k) == v
+        await session.close()
+
     def test_h2h_league(self, loop, h2h_league):
-        assert h2h_league.__str__() == "League 760869 - 760869"
+        assert h2h_league.__str__() == "League 829116 - 829116"
+
+    async def test_get_fixtures_with_known_gameweek_unauthorized(
+        self, loop, h2h_league):
```
Comment (amosbastian) [inline tests/test_h2h_league.py]: Should be indented for readability, I think.

File: tests/test_user.py [hunk 4]
```diff
@@ -1,62 +1,358 @@
...
+        ],
+        "h2h": [
+
+        ],
+        "classic": []
+    }
+}
+
+
+@pytest.fixture()
```
Comment (amosbastian) [inline tests/test_user.py]: Also in `conftest.py`.

File: tests/test_fixture.py (hunk 1 above)
Comment (espoem) [reply tests/test_fixture.py]: conftest.py is intended for shared fixtures. However, I found that they are independent and thus it made more sense to me to move it to respective class test file. Moreover, when testing the initialization it reduces redundancy of test data as it does not need to be repeated nor imported.

I can change it but I think it is not necessary to put all fixtures to common file if they are not shared across different tests.

File: tests/test_h2h_league.py (hunk 3 above)
Comment (espoem) [reply tests/test_h2h_league.py]: Indented how?

**Pull request merged**